src/ast.rs

```diff
@@ -148,3 +150,79 @@ fn concat(l: Box<Expression>, op: ConcatOp, r: Box<Expression>) -> Expression {
     v.reverse();
     Expression::Sequence(v, op)
 }
+
+const MAX_DEPTH: usize = 10;
+impl<'a> Arbitrary<'a> for BinaryOp {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(u.choose(&[
+            BinaryOp::Add,
+            BinaryOp::Sub,
+            BinaryOp::Mul,
+            BinaryOp::Div,
+            BinaryOp::Eq,
+            BinaryOp::Ne,
+            BinaryOp::Lt,
+            BinaryOp::Le,
+            BinaryOp::Gt,
+            BinaryOp::Ge,
+            BinaryOp::Contain,
+            BinaryOp::And,
+            BinaryOp::Or,
+            BinaryOp::Exp,
+        ])?
+        .clone())
+    }
+}
+
+impl<'a> Arbitrary<'a> for ConcatOp {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(u.choose(&[ConcatOp::Add, ConcatOp::Sub])?.clone())
+    }
+}
+
+impl<'a> Arbitrary<'a> for UnaryOp {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(u.choose(&[UnaryOp::Not, UnaryOp::Neg])?.clone())
+    }
+}
+
+impl<'a> Arbitrary<'a> for Expression {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        arbitrary_expr(u, 0)
+    }
+}
+
+fn arbitrary_expr(u: &mut Unstructured<'_>, depth: usize) -> arbitrary::Result<Expression> {
+    if depth >= MAX_DEPTH {
```

Comment:
So my earlier point was I don't think you need this if and match expression at all, just limit the range of the one on line 208.